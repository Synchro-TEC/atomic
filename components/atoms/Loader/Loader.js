import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Loader = ({ type, large, ...otherProps }) => {
  const baseClass = `sv-${type}-loader`;
  const rootClass = cx(baseClass, { large });

  return <span className={rootClass} {...otherProps} />;
};

Loader.defaultProps = {
  type: 'spin',
};

Loader.propTypes = {
  large: PropTypes.bool,
  type: PropTypes.oneOf(['bar', 'spin']),
};

export default Loader;
