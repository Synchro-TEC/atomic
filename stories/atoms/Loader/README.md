The `<Loader>` component is a loader.