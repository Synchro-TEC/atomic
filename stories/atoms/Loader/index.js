import React from 'react';
import { boolean } from '@storybook/addon-knobs';
import { optionalSelect } from '../../../utils/optionalSelect';

import readme from './README.md';
import Loader from '../../../components/atoms/Loader/Loader';

const options = {
  spin: 'spin',
  bar: 'bar',
};

const component = () => <Loader type={optionalSelect('Type', options, '')} large={boolean('Large', false)} />;

export default [readme, component];
