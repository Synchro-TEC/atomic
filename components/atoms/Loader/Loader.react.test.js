import React from 'react';
import Loader from './Loader';

describe('Loader Component', () => {
  test('Render Loader Component', () => {
    const loader = shallow(<Loader />);
    expect(loader).toMatchSnapshot();
  });
});
