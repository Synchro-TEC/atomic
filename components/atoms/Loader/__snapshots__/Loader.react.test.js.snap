// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component Render Loader Component 1`] = `
<span
  className="sv-spin-loader"
/>
`;
